### DISCUSSION TITLE
what is namespace for .Else()

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5593] [Q&A]
Getting below error for Else() condition in elsa 3.0
Error	CS0103	The name 'Else' does not exist in the current context	Rak.Motori.MS.ElsaWorkflow.Application	C:\MotoriCode\elsaworkflowmicroservice\src\Rak.Motori.MS.ElsaWorkflow.Application\Workflows\VehicleOwnershipRenewalWorkflow.cs	18	Active

public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
            .Then<Activity2>()
            // Add a conditional activity here
            .If(context => context.GetInput<bool>("Flag"))
                .Then<Activity3>() // This will execute if the flag is true
            **.Else()**
                .Then<Activity4>() // This will execute if the flag is false
            .Then<Activity4>(); // Continue with Activity4 after Activity3 if the flag was true
    }
}

**Comment by glime-ai** [comment]
Based on the context provided, it seems like you are trying to define a conditional flow within an Elsa 3.0 workflow using the `If` and `Else` constructs. However, you are encountering a compilation error stating that 'Else' does not exist in the current context.

In Elsa 3.0, the way to define conditional flows has changed from previous versions. The `Else` method might not be directly available or might be used incorrectly in your code. Instead, you should use the `When` method to specify the outcome branches for the `If` activity.

Here's an example of how you can define an `If` activity with `Then` and `Else` branches in Elsa 3.0:

```csharp
public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
            .Then<Activity2>()
            // Add a conditional activity here
            .Add<If>(activity => activity
                .WithCondition(context => context.GetInput<bool>("Flag"))
                .When(OutcomeNames.True).Then<Activity3>() // This will execute if the flag is true
                .When(OutcomeNames.False).Then<Activity4>()) // This will execute if the flag is false
            .Then<Activity5>(); // Continue with Activity5 after the conditional branch
    }
}
```

In this example, `OutcomeNames.True` and `OutcomeNames.False` are used to define the branches for the `If` activity. The `Then` method is used to specify the next activity to execute for each branch.

Please note that the example above assumes that `Activity3` and `Activity4` are mutually exclusive based on the condition. After the conditional execution, the workflow continues with `Activity5`.

Make sure to replace `Activity1`, `Activity2`, `Activity3`, `Activity4`, and `Activity5` with the actual activities you want to use in your workflow.

If you need further assistance, please provide more details about the error and the context in which you are using the `If` activity.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by NiveditaGalwade** [comment]
getting below error 
Error	CS1061	'ISetupActivity<If>' does not contain a definition for 'When' and no accessible extension method 'When' accepting a first argument of type 'ISetupActivity<If>' could be found (are you missing a using directive or an assembly reference?)
